tooling/encryption.md

```diff
@@ -0,0 +1,206 @@
+
+To securely share secrets with others, for example to send service account credentials with a colleague on your team who does not use a [password manager](https://bitwarden.com), you can use PGP.
+
+### Create a PGP Key Pair
+Before you can send or receive PGP messages, you need to create a key pair. You can do this with [GnuPG](https://gnupg.org).
+1. Install `gpg` if you don't already have it.
+    - On Debian-family Linux:
+        ```bash
+        sudo apt-get install -y gpg
+        ```
+    - On macOS, you can install it using [Homebrew](https://brew.sh):
+        ```bash
+        brew install gnupg
+        ```
+    - On Windows, you can install it using [Chocolatey](https://chocolatey.org):
+        ```powershell
+        choco install gpg
+        ```
+1. Use your [password manager](https://bitwarden.com) to generate a strong, unique passphrase for your key pair. You will need this passphrase to decrypt messages.
+1. Generate a new key pair.
+    > When you run this next command, it will ask you for a passphrase to protect your key...and you will not be able to click outside the dialogue box. So, if you use a password manager, copy your passphrase to your clipboard before running this command.
+    ```bash
+    gpg --batch --gen-key <<TXT
+    Key-Type: RSA
+    Key-Length: 4096
+    Subkey-Type: RSA
+    Subkey-Length: 4096
+    Name-Real: Bill Gates
+    Name-Email: someone@example.com
+    Expire-Date: 2024-10-01
+    TXT
+    ```
+    If that command doesn't work for you then you can generate a key pair manually by answering some questions. Pick either "RSA and RSA" and use a key length of at least 2048 bits, or pick "ECC and ECC" and use `ed25519` or any `NSIT` curve with a key length of at least 256 bits.
+    ```bash
+    gpg --full-generate-key
+    ```
+    On the last step, be sure to press `o` for "okay" to generate the key pair.
+
+### View Your Key Pair
+You can see what key pairs you have on your computer using this command.
+```bash
+gpg --list-keys
+```
+
+### Share Your Public Key
+You need to share your public key to receive encrypted messages. This command will print it out:
+```bash
+gpg --armor --export someone@example.com
+```
+You can then copy and paste the output into an email or IM, including the `-----BEGIN PGP PUBLIC KEY BLOCK-----` and `-----END PGP PUBLIC KEY BLOCK-----` lines. Alternatively, you can save it to a file and attach that to your message.
+```bash
+gpg --armor --export someone@example.com > public-key.asc
+```
+
+### Import a Public Key
+If someone sends you their public key, you can import it using this command:
+```bash
+gpg --import public-key.asc
+```
+Or, if you are copying and pasting it from an email or IM, you can do this:
+```bash
+gpg --import <<TXT
+```
+Then paste the key block, and finish with:
+```bash
+TXT
+```
+You will see their name, email, and key here:
+```bash
+gpg --list-keys
+```
+
+### Encrypt a Message
```

Comment:
A couple other useful tips for basic encryption that aren't mentioned in here:
1. It's possible to encrypt a single message to multiple recipients.
2. It's possible to sign and encrypt at the same time, to provide some authenticity on where the encrypted message came from.

Reply:
Both added, with signing now being the default in my examples as it is good practice.